### DISCUSSION TITLE
LoadDocument in Index without knowing the exact CollectioName (only the Id)

### DISCUSSION THREAD

**Discussion by oiBio** [ravendb/ravendb#12895] [Q&A] [answered]
Hello,

i have a collection that contains a list of entity ids from different colellections. all listed entities have some common fields (like name) and i want to query them all.

it looks like this

```
              {
                    OtherIds = new[]
                    {
                        "Vakanz/3",
                        "Vakanz/2",
                        "EntityTest2/1",
                        "EntityTest2/2",
                        "AnotherEntity/123"
                    }
                }
```

Now i want an index that loads each of the "OtherIds" and make the Name field searchable. Unfortunatly i do not have the collection name for the entity, only the id.

   ```
     private class PrivateHelperClassForName
        {
            public string Name { get; set; }
        }

        public HasNameIndex()
        {
            AddMap<ParentEntity>(parents=> from v in parents
                                               let entity = LoadDocument<PrivateHelperClassForName>(v.OtherIds)
                                               select new
                                               {
                                                   Id = v.Id,
                                                   Searchfield = entity.Select(e => new { e.Name })
                                               });
            Index(e => e.Searchfield, FieldIndexing.Search);
        }
```

im pretty sure this worked in the past, but now i think i need the concrete collectionnames to get this working.

Is there a workaround for this scenario?

**Comment by ayende** [comment, marked as answer]
Yes, in 4.x we made the collection of the loaded document mandatory.
You can chain `LoadDocument`s calls, like so:

                                            let entity = LoadDocument<Vakanz>(v.Id) ?? LoadDocument<Other>(v.Id)

**Reply by oiBio** [reply]
Hi Oren, the problem is, that i have mutliple Id´s in the field and these can be from many different entities

**Reply by ayende** [reply]
It is a bit more awkward in that case, but you can use:

```
let items = v.ListOfIds.Select(id => LoadDocument<OptionA>(id) ?? LoadDocument<OptionB>(id) )
```

The reason for this limit is that we *must* know what collections we are working on to load documents, that is important for a bunch of optimizations on indexing speed and performance of the indexing in this scenario.

**Reply by oiBio** [reply]
This is not the answer is was hoping for... i think we need to rethink our index/implementation. 
Maybe something like an common interface and and artificial index or something in this direction.

anyways thank you for your fast respose!
